user.go

```diff
@@ -2,27 +2,100 @@ package main
 
 import (
 	"github.com/gorilla/mux"
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/mongodb/mongo-go-driver/bson/primitive"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	"github.com/mongodb/mongo-go-driver/mongo/options"
 	"net/http"
-	"fmt"
 	"encoding/json"
+	"context"
 )
 
 // User is a structure used for serializing/deserialzing user data
 type User struct {
-	ID string `json:"id"`
+	ID primitive.ObjectID `json:"_id" bson:"_id"`
+	Name string `json:"name" bson:"name"`
+	Admin bool `json:"admin" bson:"admin"`
 }
 
 // UserHandler is a structure used for handling requests for user related actions
-type UserHandler struct {}
+type UserHandler struct {
+	Collection *mongo.Collection
+}
+
+// GetUsersHandler Gets a GET request with and responds with the
+// list of all users' data from the database
+func (h UserHandler) GetUsersHandler(w http.ResponseWriter, r *http.Request) {
+	findOptions := options.Find()
+	filter := bson.D{}
+	var users []*User
+
+	cur, err := h.Collection.Find(context.Background(), filter, findOptions)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	for cur.Next(context.Background()) {
+		var user User
+		err := cur.Decode(&user)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		users = append(users, &user)
+	}
+
+	if err := cur.Err(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = cur.Close(context.Background())
```

Comment:
wont you prefer to `defer cur.Close(context.Background())`?

Reply:
Thought about it but where I copied this from used it like this, I could add defer

Reply:
`defer cur.Close(context.Background())`

[From here](https://godoc.org/github.com/mongodb/mongo-go-driver/mongo)